src/library/Uno.Material/Controls/Chips/ChipGroup.cs

```diff
@@ -0,0 +1,332 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Uno.Extensions.Specialized;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Markup;
+
+namespace Uno.Material.Controls
+{
+	public partial class ChipGroup : ItemsControl
+	{
+		public event ChipItemEventHandler ItemClick;
+		public event ChipItemEventHandler ItemChecked;
+		public event ChipItemEventHandler ItemUnchecked;
+
+		private bool _isSynchronizingSelection = false;
+		private bool _isUpdatingSelection = false;
+
+		public ChipGroup()
+		{
+			RegisterPropertyChangedCallback(ItemsSourceProperty, (s, e) => (s as ChipGroup)?.OnItemsSourceChanged());
+
+			this.Loaded += OnLoaded;
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+			ApplyThumbnailTemplate();
+		}
+
+		private void OnLoaded(object sender, RoutedEventArgs e)
+		{
+			EnforceSelectionMode();
+		}
+
+		private void OnSelectionMemberPathChanged(DependencyPropertyChangedEventArgs e)
+		{
+			var binding = SelectionMemberPath != null
```

Comment:
why not 
if(SelectionMemberPath == null) 
return;
then you don't have to null check in your foreach

Reply:
We still need to iterate through each child even if `binding` is null. `container.ClearValue()`